src/main/java/amidst/fragment/Fragment.java

```diff
@@ -18,26 +19,29 @@
 
 /**
  * This class contains nearly no logic but only simple and atomic getters and
- * setters.
- *
+ * setters. </br>
+ * </br>
  * The life-cycle of a Fragment is quite complex to prevent the garbage
  * collection from running too often. When a fragment is no longer needed it
  * will be kept available in a queue, so it can be reused later on. The
- * life-cycle consists of the two flags isInitialized and isLoaded.
- * setInitialized(true) can be called from any thread, however
- * setInitialized(false), setLoaded(true) and setLoaded(false) will always be
- * called from the fragment loading thread, to ensure consistent state. Also,
- * setInitialized(true) will only be called again after setInitialized(false)
- * was called. It is not possible that isLoaded is true while isInitialized is
- * false.
- *
+ * life-cycle consists of the three flags: isInitialized, isLoading, and
+ * isLoaded. isInitialized can be set to true from any thread, however setting
+ * isInitialized to false as well as any modification to isLoading and isLoaded
+ * will always be called from the fragment loading thread, to ensure a
+ * consistent state. Also, isInitialized will only be set to true again after it
+ * was set to false. It is not possible that isLoading or isLoaded is true while
+ * isInitialized is false. It is also not possible for isLoaded to be true while
+ * isLoading is true. </br>
+ * </br>
```

Comment:
This feels quite confusing, and I think these three flags actually want to be a state machine.

From your description, there are only four valid states:
 - `!isInitialized && !isLoading && !isLoaded`; the fragment is uninitialized
 - `isInitialized && !isLoading && !isLoaded`; the fragment is waiting to be loaded
 - `isInitialized && isLoading && !isLoaded`; the fragment is loading
 - `isInitialized && isLoading && isLoaded`; the fragment is fully loaded

It is possible to regroup the three flags into a single `state` field, which can take one of the four values given above?

Reply:
This has been fixed by creating a new enum in the `Fragment` class called `State`. There are now four different states: `UNINITIALIZED`, `INITIALIZED`, `LOADING`, and `LOADED`.